addons/mail/static/src/models/dialog/dialog.js

```diff
@@ -19,15 +19,26 @@ function factory(dependencies) {
             if (this.followerSubtypeList) {
                 return replace(this.followerSubtypeList);
             }
+        },
+        _computeComponentName() {
+            if (this.attachmentViewer) {
+                return 'AttachmentViewer';
+            }
+            if (this.followerSubtypeList) {
+                return 'FollowerSubtypeList';
+            }
+            return '';
         }
-    }
-
-    Dialog.fields = {
+    },
+    fields: {
         attachmentViewer: one2one('mail.attachment_viewer', {
             isCausal: true,
             inverse: 'dialog',
             readonly: true,
         }),
+        componentName: attr({
+            compute: '_computeComponentName',
+        }),
```

Comment:
Would be nice to isolate this change in a commit just before the new syntax (introduce `componentName` and use it in `DialogComponent` template)